baybe/surrogates/base.py

```diff
@@ -47,6 +52,16 @@
 """
 
 
+class _NoTransform(Enum):
+    """Sentinel class."""
+
+    IDENTITY_TRANSFORM = auto()
+
+
+_IDENTITY_TRANSFORM = _NoTransform.IDENTITY_TRANSFORM
```

Comment:
why is this needed and cannot simply be handled by `None` ?

Reply:
It could but I don't think it would be very clean. The problem is that the `_output_scaler` variable is `None` when not yet fitted, so setting it also to `None` after fitting to indicate that identity transform should be used would be very ambiguous. That's exactly the case for sentinels. Of course, we could also imagine maintaining a single generic sentinel somewhere centrally, but since everything private anyway, I didn't bother too much

Reply:
ok isee, but for the input scaling this dilemma doesnt apply right? because there wenow have the none return for identity

Reply:
Correct, for the input scaling the problem does not arise because there it is sufficient to only store the composite transformer, which is always a column transformer. For the output, we however need to also store the scaler separately in addition to the entire transform pipeline because it is later required for the posterior transform. And that part causes the problem because it can potentially be "no transform" (currently represented through the sentinel).

Could imagine that we can find a less heavy approach later (e.g. I can imagine that we can even outsource part of our current processing to botorch, e.g. the scalarization of the desirability objective etc). But for now I think the basic functionality is in focus. Deliberately kept everything private where it was possible, so we can refine if needed in later versions.